### DISCUSSION TITLE
Illuminator version 3.0-alpha

### DISCUSSION THREAD

**Discussion by manuGil** [Illuminator-team/Illuminator#83] [Q&A]
Ask questions about use, implementation or development of the 3.0-alpha version.

**Comment by manuGil** [comment]
# Deprecated `results`

The keyword `results` has been removed from the `scenario` section of the scenario config YAML file because it was redundant. The `file` keyword  in `monitor` has the same purpose.

**Reply by manuGil** [reply]
After this was removed from the schema, a few bugs appeared on  the CLI component. They have been fixed in #84

**Comment by manuGil** [comment]
# Python Interface

Now, it is possible to run a simulation within Python as follows:
#85 

```python
from illuminator.engine import Simulation

simulation = Simulation('<path/to/config.yaml>')
simulation.run()
```

**Comment by jgrguric96** [comment]
# Proposed ModelConstructor fixes
Made a PR with some proposed changes. I do not think they should be accepted yet, but it is a nice step in the right direction.

### NOTE:
This is a Copy paste of my PR documentation

### **Fixed:**

- Missing cluster import (temporary)
- Missing parameter in parent class ModelConstructor
- Incorrect sim_params length (this can be deleted since its unused)
- Incorrect number of parameters used for .create(...) functions.

### **Added:**

- Global dictionary current_model added to engine.py (alternative should be found and replace this implementation)
-- This variable is called when the mosaik.world.start is creating the appropriate child model
- Alternative (commented out) regex for yaml files. It might be useful later on during connection stage fixes.
- Alternative (commented out) naming convention for connections within adder_copy.yaml
- Proposed (within comments) fix for adding custom number of parameters for create function

### **TODO**
- ~Currently this implementation now enters the "create" stage of the simulation, and crashes due to unknown parameters.~
- Current implementation creates models as expected, however it fails on creating connections between them. Likely issue is invalid naming convention

**Comment by manuGil** [comment]
@JortGroen  and @Eutardigrada  
I looked into the monitor issue the CLI. As you mentioned the CLI throws an error about a missing `collector`.  However, the reason for that is that when we merged the code, some of the old code and the new code were kept. In the new implementation the Collector/Monitor is implemented in `Simulation.run()`. Therefore, the lines calling the collector in `cli/main.py` must be removed:

https://github.com/Illuminator-team/Illuminator/blob/1c120214553180da19ef05910a4a2ff0b264fe79/src/illuminator/cli/main.py#L32-L54

Removing those line remove the error; however, notice that the `simple_test.yaml` simulation does not succeed because of the 'general issue' with connecting the *scenario API* with *Mosaik*.
